launch/launch/launch_service.py

```diff
@@ -234,7 +248,9 @@ def run(self, *, shutdown_when_idle=True) -> None:
 
         # Acquire the lock and initialize the asyncio loop.
         with self.__loop_from_run_thread_lock:
+            global _g_loops_used
             self.__loop_from_run_thread = osrf_pycommon.process_utils.get_loop()
+            _g_loops_used.add(self.__loop_from_run_thread)
```

Comment:
maybe better to move this line after the next check for `None` in case someone catches the exception and we end up trying to close a `None` loop. (doubt it would happen but seems more defensive)